### DISCUSSION TITLE
Return the FTP server

### DISCUSSION THREAD

**Discussion by Alex11381** [pacien/tincapp#129] [Ideas]
Hello. Could you, in future versions of the application, return the functionality of accessing configuration files only via ftp. Or make a separate branch with this function. I consider ftp access as it was implemented before version 0.38 to be more secure.

**Comment by pacien** [comment]
Hi,

The embedded FTP server was using a not-so-maintained library.  Transferring
configuration files through a possibly unencrypted connection, as well as
opening a port on the mobile device, were also both questionable regarding
security.  It was quite unreliable depending on the device too.

I'm currently investigating other ways to expose the configuration files
more reliably on all devices, since the shared directory doesn't seem to
be accessible on all vendor variants of Android.

In the meantime, you can still download an older version
of the app, containing the embedded FTP server here:
https://tincapp.pacien.org/downloads/tincapp.0.37.apk

I might re-introduce it in later versions if I cannot find another way.
However I'd like to avoid that for the reasons explained above.

**Reply by Alex11381** [reply]
However, this method made it possible to hide configuration files, including the key, from the user. And I always turned off ftp after setting it up. In general, in my case, the user did not have access to the ability to enable ftp and configuration files. Thank you, I do, I use version 0.37, but I would like to receive software updates and still maintain ftp access. I will be glad if this functionality returns later.
